sql/index/pilosa/driver.go

```diff
@@ -0,0 +1,213 @@
+package pilosa
+
+import (
+	"context"
+	"crypto/sha1"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	pilosa "github.com/pilosa/go-pilosa"
+	"gopkg.in/src-d/go-mysql-server.v0/sql"
+	"gopkg.in/src-d/go-mysql-server.v0/sql/index"
+)
+
+const (
+	// DriverID the unique name of the pilosa driver.
+	DriverID = "pilosa"
+	// IndexNamePrefix the pilosa's indexes prefix
+	IndexNamePrefix = "idx"
+	// FrameNamePrefix the pilosa's frames prefix
+	FrameNamePrefix = "frm"
+)
+
+// Driver implements sql.IndexDriver interface.
+type Driver struct {
+	root   string
+	client *pilosa.Client
+}
+
+// NewDriver returns a new instance of pilosa.Driver
+// which satisfies sql.IndexDriver interface
+func NewDriver(root string, client *pilosa.Client) *Driver {
+	return &Driver{
+		root:   root,
+		client: client,
+	}
+}
+
+// NewIndexDriver returns a default instance of pilosa.Driver
+func NewIndexDriver(root string) sql.IndexDriver {
+	return NewDriver(root, pilosa.DefaultClient())
+}
+
+// ID returns the unique name of the driver.
+func (*Driver) ID() string {
+	return DriverID
+}
+
+// Create a new index.
+func (d *Driver) Create(db, table, id string, expr []sql.ExpressionHash, config map[string]string) (sql.Index, error) {
+	path, err := mkdir(d.root, db, table, id)
+	if err != nil {
+		return nil, err
+	}
+
+	cfg := index.NewConfig(db, table, id, expr, d.ID(), config)
+	err = index.WriteConfigFile(path, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &sqlIndex{
+		path:        path,
+		client:      d.client,
+		db:          db,
+		table:       table,
+		id:          id,
+		expressions: expr,
+	}, nil
+}
+
+// LoadAll loads all indexes for given db and table
+func (d *Driver) LoadAll(db, table string) ([]sql.Index, error) {
+	root := filepath.Join(d.root, db, table)
+
+	var indexes []sql.Index
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
```

Comment:
should we just skip the directory? maybe that error is a permission denied error or something similar.

Reply:
What we can do (except logging)?
- Stop loading all indexes
- Return accumulated error with all dirs which cannot be loaded

Reply:
Yep, we should handle that error instead of just skipping it